gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -66,10 +66,12 @@
  * http header of requests to the service.
  */
 @InternalExtensionOnly
-public final class InstantiatingGrpcChannelProvider implements TransportChannelProvider {
+public final class InstantiatingGrpcChannelProvider
+    implements TransportChannelProvider, ChannelFactory {
```

Comment:
I would prefer avoid extending the public surface to ChannelFactory. Can you implement it as an inner class instead?